packages/compartment-mapper/src/import.js

```diff
@@ -38,17 +43,36 @@ const assignParserForLanguage = (options = {}) => {
 };
 
 /**
+ * @overload
+ * @param {SyncReadPowers} readPowers
+ * @param {string} moduleLocation
+ * @param {SyncArchiveOptions} options
+ * @returns {Promise<Application>}
+ */
+
+/**
+ * @overload
  * @param {ReadFn | ReadPowers} readPowers
  * @param {string} moduleLocation
- * @param {ArchiveOptions} [options]
+ * @param {LoadLocationOptions} [options]
+ * @returns {Promise<Application>}
+ */
+
+/**
+ * @param {ReadFn|ReadPowers|SyncReadPowers} readPowers
+ * @param {string} moduleLocation
+ * @param {LoadLocationOptions} [options]
  * @returns {Promise<Application>}
  */
 export const loadLocation = async (
   readPowers,
   moduleLocation,
   options = {},
 ) => {
-  const { dev, tags, conditions = tags, commonDependencies, policy } = options;
+  const { dev, tags, commonDependencies, policy } = options;
+  // conditions are not present in SyncArchiveOptions
+  const conditions =
+    'conditions' in options ? options.conditions || tags : tags;
```

Comment:
```suggestion
  const { dev, tags, conditions = tags, commonDependencies, policy } = options;
```

Reply:
I bet typescript made him do it.

Reply:
yes, it would require a type assertion otherwise

Reply:
In that case, disregard. I wonder why I got away with this pattern elsewhere.